dashboard/api_test.py

```diff
@@ -735,6 +735,68 @@ def mocked_get_status_for_courserun(run, enrollments):  # pylint: disable=unused
         mock_format.assert_called_once_with(run1, api.CourseStatus.OFFERED, None, position=1)
 
 
+class UserProgramInfoIntegrationTest(ESTestCase):
+    """Integration tests for get_user_program_info"""
+    @classmethod
+    def setUpTestData(cls):
+        super(UserProgramInfoIntegrationTest, cls).setUpTestData()
+        cls.user = UserFactory()
+        # create the programs
+        cls.program_non_fin_aid = ProgramFactory.create(full=True, live=True)
+        cls.program_fin_aid = ProgramFactory.create(full=True, live=True, financial_aid_availability=True)
+        cls.program_unenrolled = ProgramFactory.create(full=True, live=True)
+        cls.program_not_live = ProgramFactory.create(live=False)
+        for program in [cls.program_non_fin_aid, cls.program_fin_aid, cls.program_not_live]:
+            models.ProgramEnrollment.objects.create(user=cls.user, program=program)
+
+    def setUp(self):
+        super().setUp()
+        self.expected_programs = [self.program_non_fin_aid, self.program_fin_aid]
+        self.edx_client = MagicMock()
+
+    @patch('dashboard.api.get_student_enrollments', autospec=True)
+    @patch('dashboard.api.get_student_certificates', autospec=True)
+    @patch('dashboard.api.get_student_current_grades', autospec=True)
+    def test_format(self, mock_grades, mock_cert, mock_enr):
+        """Test that get_user_program_info fetches edx data and returns a list of Program data"""
+        result = api.get_user_program_info(self.user, self.edx_client)
+
+        for mocked_edx_get_function in [mock_grades, mock_cert, mock_enr]:
+            assert mocked_edx_get_function.call_count == 1
+            assert mocked_edx_get_function.call_args[0] == (self.user, self.edx_client)
+
+        assert len(result) == 2
+        for i in range(2):
+            expected = {
+                "id": self.expected_programs[i].id,
+                "description": self.expected_programs[i].description,
+                "title": self.expected_programs[i].title,
+                "financial_aid_availability": self.expected_programs[i].financial_aid_availability,
+            }
+            assert is_subset_dict(expected, result[i])
+
+    def test_past_course_runs(self):
+        """Test that past course runs are returned in the API results"""
+        # Set a course run to be failed
+        program = self.program_non_fin_aid
+        course = program.course_set.first()
+        failed_course_run = \
+            set_course_to_failed(user=self.user, course=course)  # pylint: disable=unexpected-keyword-arg
```

Comment:
that method (and several others in the module) has decorator that takes a course kwarg, fetches a 'default' course run, scraps the course kwarg, and inserts the course_run kwarg. the kwarg setup is a bit of an anti-pattern - the alter_data commands were written with a little less attention to standards. anyway, it works for now. i'm going to open up and issue to move to some more formal integration tests for the dashboard API. part of that issue will entail cleaning this up this code and making sure it's used properly in test cases